var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -62,6 +62,21 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.18:", type="build")
 
+    # Profiling
+    depends_on(
+        "hiop+deepchecking build_type=RelWithDebInfo", when="+hiop build_type=RelWithDebInfo"
+    )
+    depends_on("hiop~deepchecking  build_type=Release ", when="+hiop build_type=Release ")
+
+    # Control the package's build-type depending on the release or debug flag
+    for pkg in ["raja", "umpire", "magma", "camp", "ginkgo"]:
+        depends_on("{0} build_type=Release".format(pkg), when="build_type=Release")
+        depends_on("{0} build_type=RelWithDebInfo".format(pkg), when="build_type=RelWithDebInfo")
```

Comment:
```suggestion
        depends_on("{0} build_type=Release".format(pkg), when="+{0} build_type=Release".format(pkg))
        depends_on("{0} build_type=RelWithDebInfo".format(pkg), when="+{0} build_type=RelWithDebInfo".format(pkg))
```

I'm not as adept at reading audit logs as @alalazo, but I think this is the correct logic, and a upon re-reading I think I found the bug

Reply:
Turns out there isn't actually a variant that maps directly to each package... @ryandanehy can you create a tuple list that pairs each package to the variant combination that controls it, and then modify this loop accordingly?